apps/remote_control/lib/lexical/remote_control/code_mod/rename/file.ex

```diff
@@ -0,0 +1,172 @@
+
+    with {:ok, prefix, old_module_paths} <- fetch_conventional_prefix(relative_path),
+         {:ok, new_name} <- fetch_new_name(entry, new_suffix) do
+      extname = Path.extname(entry.path)
+
+      suffix =
+        new_name
+        |> Macro.underscore()
+        |> maybe_insert_special_phoenix_folder(old_module_paths, new_name)
+
+      new_path = Path.join([root_path, prefix, "#{suffix}#{extname}"])
+
+      old_uri = Document.Path.ensure_uri(entry.path)
+      new_uri = Document.Path.ensure_uri(new_path)
+      Document.Changes.RenameFile.new(old_uri, new_uri)
+    else
+      _ -> nil
+    end
+  end
+
+  defp relative_path(path, root_path) do
+    Path.relative_to(path, root_path)
+  end
+
+  defp root_path do
+    Project.root_path(RemoteControl.get_project())
+  end
+
+  defp fetch_new_name(entry, new_suffix) do
+    uri = Document.Path.ensure_uri(entry.path)
+    text_edits = [Document.Edit.new(new_suffix, entry.range)]
+
+    with {:ok, document} <- Document.Store.open_temporary(uri),
+         {:ok, edited_document} =
+           Document.apply_content_changes(document, document.version + 1, text_edits),
+         {:ok, %{context: {:alias, alias}}} <-
+           Ast.surround_context(edited_document, entry.range.start) do
+      {:ok, to_string(alias)}
+    else
+      _ -> :error
+    end
+  end
+
+  defp fetch_conventional_prefix(path) do
+    result =
+      path
+      |> Path.split()
+      |> Enum.chunk_every(2, 2)
+      |> Enum.reduce({[], []}, fn
+        ["apps", app_name], _ ->
+          {[], [app_name, "apps"]}
+
+        ["lib", follow_element], {elements, prefix} ->
+          {[follow_element | elements], ["lib" | prefix]}
+
+        ["test", follow_element], {elements, prefix} ->
+          {[follow_element | elements], ["test" | prefix]}
+
+        remain, {elements, prefix} ->
+          {remain ++ elements, prefix}
+      end)
+
+    case result do
+      {_, []} ->
+        :error
+
+      {module_paths, prefix} ->
+        prefix = prefix |> Enum.reverse() |> Enum.join("/")
+        {:ok, prefix, module_paths}
+    end
+  end
+
+  defp maybe_insert_special_phoenix_folder(suffix, old_module_paths, new_name) do
+    web_app? = new_name |> String.split(".") |> hd() |> String.ends_with?("Web")
+
+    insertions =
+      cond do
+        not web_app? ->
+          ""
+
+        phoenix_component?(old_module_paths, new_name) ->
+          "components"
+
+        phoenix_controller?(old_module_paths, new_name) ->
+          "controllers"
+
+        phoenix_live_view?(old_module_paths, new_name) ->
+          "live"
+
+        true ->
+          ""
+      end
+
+    suffix
+    |> String.split("/")
+    |> List.insert_at(1, insertions)
+    |> Enum.reject(&(&1 == ""))
+    |> Enum.join("/")
+  end
+
+  defp phoenix_component?(old_module_paths, new_name) do
+    under_components? = "components" in old_module_paths
+    component? = String.ends_with?(new_name, ["Components", "Layouts", "Component"])
```

Comment:
i feel the same way about this heuristic as I did with controllers, there's no real indication that any of these are actually components, layouts or controllers. 
You can definitely check the modules though